opendc-model-odc/jpa/src/main/resources/jpa/schema.xml

```diff
@@ -191,6 +191,18 @@
                     <cascade-persist/>
                 </cascade>
             </many-to-many>
+            <many-to-many name="dependents" target-entity="Task">
```

Comment:
@fabianishere I think I broke something here. I think the dependents are now also being inserted into the table, which conflicts with already dependencies ? I've attached the log.


```log
2018-12-19 14:56:27.733 [main] ERROR org.hibernate.engine.jdbc.batch.internal.BatchingBatch - HHH000315: Exception executing batch [org.h2.jdbc.JdbcBatchUpdateException: Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 21, 21)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]], SQL: insert into task_dependencies (first_task_id, second_task_id) values (?, ?)
2018-12-19 14:56:27.734 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.734 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 21, 21)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.734 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.734 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 21, 21)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.735 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.735 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 10, 20)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.737 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.737 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 11, 19)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.737 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.737 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 12, 18)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.737 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.738 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 1, 17)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.738 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.738 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 13, 16)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.738 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.738 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 2, 15)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.738 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.738 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 20, 14)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.739 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.739 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 18, 13)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.739 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.739 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 7, 12)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.739 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.739 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 19, 11)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.739 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.739 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 8, 10)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.740 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.740 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 9, 9)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.740 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.740 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 14, 8)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.740 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.740 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 3, 7)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.741 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.741 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 15, 6)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.741 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.741 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 4, 5)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.741 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.741 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 16, 4)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.741 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.741 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 5, 3)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.742 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.742 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 17, 2)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.742 [main] WARN  org.hibernate.engine.jdbc.spi.SqlExceptionHelper - SQL Error: 23505, SQLState: 23505
2018-12-19 14:56:27.742 [main] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Unique index or primary key violation: "PRIMARY_KEY_6B ON PUBLIC.""task_dependencies""(""second_task_id"", ""first_task_id"") VALUES (0, 6, 1)"; SQL statement:
insert into task_dependencies (first_task_id, second_task_id) values (?, ?) [23505-197]
2018-12-19 14:56:27.746 [main] ERROR org.hibernate.internal.ExceptionMapperStandardImpl - HHH000346: Error during managed flush [org.hibernate.exception.ConstraintViolationException: could not execute batch]
2018-12-19 14:56:28.124 [main] INFO  org.hibernate.tool.schema.internal.SchemaDropperImpl$DelayedDropActionImpl - HHH000477: Starting delayed drop of schema as part of SessionFactory shut-down'
2018-12-19 14:56:28.124 [main] INFO  org.hibernate.orm.connections.access - HHH10001501: Connection obtained from JdbcConnectionAccess [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator$ConnectionProviderJdbcConnectionAccess@59c68ed3] for (non-JTA) DDL execution was not in auto-commit mode; the Connection 'local transaction' will be committed and the Connection will be set into auto-commit mode.
2018-12-19 14:56:28.136 [main] INFO  org.hibernate.orm.connections.pooling - HHH10001008: Cleaning up connection pool [jdbc:h2:mem:opendc;DATABASE_TO_UPPER=false;]
```